rfcs/migrate-to-async-hyper.md

```diff
@@ -0,0 +1,47 @@
+# Migrating docs.rs to asynchronous code and hyper
+
+## Approach
+
+1. Update `tokio`, `futures` and `rusoto_{s3, core, credential}` to their latest versions
+
+2. Make `main` an `async fn` with `#[tokio::main]` to start the tokio runtime, add the `hyper` dependency and migrate the side services to async (By side services I mean the ones in the daemon)
```

Comment:
Why should main be async? It seems to me the benefit is coming from web requests/io be async, but there's very little io in main. It spawns worker threads if it needs to do anything heavy.

Reply:
It's a transitive thing, async functions call async functions. The idea is that all spawned tasks are part of the same runtime which makes the most efficient use of the system.

Reply:
I don't see why it needs to be async to call async functions, we already use `spawn_blocking` in a few places. My concern is this adds overhead for interactive usage since we have to start the whole tokio runtime before processing CLI args.

Reply:
Testing in debug mode with hello world the overhead of `tokio::main` is <4ms on some not great hardware.